### PR TITLE
Issue/1403 corrupt zip tests

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
NA
#### The solution
* Small refactor of file related methods to facilitate testing
* Add first tests to FileDataRepository and FileDataSource
More tests will be added in next PR
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        verify(mockFileDataSource, times(1))!!.moveZipFiles()
+        verify(mockFileDataSource, times(1))!!.moveMediaFiles()
+        observer.assertNoErrors()
+    }
+
+    @Test
+    fun moveFilesShouldFailIfMoveZipFilesFails() {
+        val exception = Exception("test")
+        `when`(mockFileDataSource!!.moveZipFiles()).thenReturn(Completable.error(exception))
+        `when`(mockFileDataSource!!.moveMediaFiles()).thenReturn(Completable.complete())
```
Comment (muloem) [inline data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt]: A little detail here but if the focus is on zip files, maybe this one doesnt need to be in there?

File: data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+        verify(mockFileDataSource, times(1))!!.moveZipFiles()
+        verify(mockFileDataSource, times(1))!!.moveMediaFiles()
+        observer.assertError(exception)
+    }
+
+    @Test
+    fun moveFilesShouldFailIfMoveMediaFilesFails() {
+        val exception = Exception("test")
+        `when`(mockFileDataSource!!.moveZipFiles()).thenReturn(Completable.complete())
```
Comment (muloem) [inline data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt]: Same here, if the focus is on media files then maybe zip files dont need to be in here?

File: data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt (hunk 1 above)
Comment (valllllll2000) [reply data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt]: if I remove the above line the test will fail as this method is called

File: data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt (hunk 2 above)
Comment (valllllll2000) [reply data/src/test/java/org/akvo/flow/data/repository/FileDataRepositoryTest.kt]: in this case you are right but on the other hand the test if for the whole class and one of it's methods is that one plus I do not think I will have the occasion to revisit those tests again

**Pull request merged**